webapp/src/components/confirm_modal.tsx

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+export type Props = {
+
+    /*
+     * Set to show modal
+     */
+    show: boolean;
+
+    /*
+     * Title to use for the modal
+     */
+    title: React.ReactNode;
+
+    /*
+     * Message to display in the body of the modal
+     */
+    message: React.ReactNode;
+
+    /*
+     * The CSS class to apply to the confirm button
+     */
+    confirmButtonClass: string;
+
+    /*
+     * The CSS class to apply to the modal
+     */
+    modalClass?: string;
+
+    /*
+     * Text/jsx element on the confirm button
+     */
+    confirmButtonText: React.ReactNode;
+
+    /*
+     * Text/jsx element on the cancel button
+     */
+    cancelButtonText: React.ReactNode;
+
+    /*
+     * Set to show checkbox
+     */
+    showCheckbox?: boolean;
+
+    /*
+     * Text/jsx element to display with the checkbox
+     */
+    checkboxText?: string;
+
+    /*
+     * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
+     */
+    onConfirm?: (arg0: boolean) => void;
+
+    /*
+     * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
+     */
+    onCancel?: (arg0: boolean) => void;
+
+    /**
+     * Function called when modal is dismissed
+     */
+    onExited?: () => void;
+
+    /*
+     * Set to hide the cancel button
+     */
+    hideCancel: boolean;
+
+}
+
+export type State = {
+}
+
+export default class ConfirmModal extends PureComponent<Props, State> {
```

Comment:
Nit: If a component has no state, then just don't pass it in here, and remove `export type State = {}` as well (Sorry for yet another requested change)
```suggestion
export default class ConfirmModal extends PureComponent<Props> {
```

Reply:
No apologies.  I really appreciate the nits too :)